webui/react/src/components/InteractiveTable.stories.tsx

```diff
@@ -28,32 +29,28 @@ const columns = new Array(20).fill(null).map(() => {
   };
 });
 
-const config: SettingsConfig = {
-  settings: [
-    {
+const config: SettingsConfig<
+  Omit<InteractiveTableSettings, 'sortDesc' | 'sortKey' | 'tableLimit' | 'tableOffset'>
+> = {
+  applicableRoutespace: 'storybook',
+  settings: {
+    columns: {
       defaultValue: columns.map((column) => column.dataIndex),
-      key: 'columns',
       storageKey: 'columns',
-      type: {
-        baseType: BaseType.String,
-        isArray: true,
-      },
+      type: array(string),
     },
-    {
+    columnWidths: {
       defaultValue: columns.map((column) => column.defaultWidth),
-      key: 'columnWidths',
       skipUrlEncoding: true,
       storageKey: 'columnWidths',
-      type: {
-        baseType: BaseType.Float,
-        isArray: true,
-      },
+      type: array(number),
     },
-    {
-      key: 'row',
-      type: { baseType: BaseType.String, isArray: true },
+    row: {
+      defaultValue: [],
+      storageKey: 'row',
```

Comment:
`storageKey` and key of object are the same, do we still need `storageKey`?
I like the new KV structure btw. its more efficient than array.